scene/2d/canvas_item.cpp

```diff
@@ -148,7 +148,7 @@ void CanvasItemMaterial::_update_shader() {
 		code += "\t\tparticle_frame = mod(particle_frame, particle_total_frames);\n";
 		code += "\t}";
 		code += "\tUV /= vec2(h_frames, v_frames);\n";
-		code += "\tUV += vec2(mod(particle_frame, h_frames) / h_frames, floor(particle_frame / h_frames) / v_frames);\n";
+		code += "\tUV += vec2(mod(particle_frame, h_frames) / h_frames, floor((particle_frame + 0.5) / h_frames) / v_frames);\n";
```

Comment:
I guess to address @lawnjelly's comment this should be:
```suggestion
		code += "\tUV += vec2(mod(floor(particle_frame + 0.5), h_frames) / h_frames, floor((particle_frame + 0.5) / h_frames) / v_frames);\n";
```

Reply:
But then this might be worth doing further up when `particle_frame` is calculated?

Maybe there's a better operator that could be done on integer values to calculate `particle_frame` in the first place instead of using floating point?

Reply:
It's kind of hard to read and follow in the stringifyed format :confused: but I think it could be:
```
floor (mod(particle_frame+0.5, h_frames)) / h_frames
```
Or do the `particle_frame += 0.5;` earlier as a once off as @akien-mga points out.

You could also _in theory_ do something like a:
```
particle_frame = round(particle_frame) + epsilon;
```
And then not need the later bodges .. but for being a little more 'clever' this is something that would be be bound to screw up on some hardware with some special precision etc.

GLES has next to no support for integers unfortunately, which makes this kind of juggling super annoying but necessary.